cloudsim-plus/src/main/java/org/cloudbus/cloudsim/power/PowerMeter.java

```diff
@@ -0,0 +1,158 @@
+package org.cloudbus.cloudsim.power;
+
+import org.cloudbus.cloudsim.core.CloudSimEntity;
+import org.cloudbus.cloudsim.core.CloudSimTags;
+import org.cloudbus.cloudsim.core.Simulation;
+import org.cloudbus.cloudsim.core.events.SimEvent;
+import org.cloudbus.cloudsim.power.models.PowerModel;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import static org.cloudbus.cloudsim.core.CloudSimTags.POWER_MEASUREMENT;
+
+/**
+ * Periodically measures the current power usage of one or more {@link PowerAware} entities,
+ * according to a defined {@link #getMeasurementInterval() interval}, storing the results.
+ *
+ * @see #getPowerMeasurements()
+ * @since CloudSim Plus 6.0.0
+ */
+public class PowerMeter extends CloudSimEntity {
+
+    private final Callable<List<? extends PowerAware<? extends PowerModel>>> powerAwareEntitiesFn;
+
+    private double measurementInterval;
+    /** @see #getStartTime() */
+    private double startTime;
+
+    private final List<PowerMeasurement> powerMeasurements = new LinkedList<>();
+
+    /**
+     * Initializes a {@link PowerMeter} to measure power consumption of a single {@link PowerAware} entity.
+     */
+    public PowerMeter(final Simulation simulation, final PowerAware<? extends PowerModel> powerAwareEntity) {
+        this(simulation, new ArrayList<>(Arrays.asList(powerAwareEntity)));
```

Comment:
Is there any reason to instantiate a `new ArrayList` here?
The `Arrays.asList` already creates a list.

Reply:
Since the asList is receiving a single argument that is converted internally to an array 
(instead of receiving an external array), the created array is encapsulated and therefore not subject
to external changes.

Reply:
You are right, thanks for catching that!